### PR TITLE
chore: buf pr checks github workflow

### LINKED ISSUE #44: PR Validation Workflow
Buf can be used to validate PRs. It should run:

- [buf lint](https://docs.buf.build/lint/usage#run-lint)
- [buf breaking change detection](https://docs.buf.build/lint/usage#run-lint)
- [buf format](https://docs.buf.build/format/usage)

### PR DESCRIPTION
**Pull request by skyerus**
## This PR

Includes PR checks workflow for buf linting, buf formatting & buf breaking changes check.

### Related Issues

Fixes #44

### CODE REVIEW

**Comment by skyerus** [comment]
The linting step demanded that `google/protobuf/empty.proto` be removed as it is an unused import.
The `buf-breaking-changes` job then fails because it thinks that removing the unused is breaking, I think we can ignore the warning and force push in this instance.

**Review by james-milligan — APPROVED** [review]

File: .github/workflows/pr-checks.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+name: "PR checks"
+
+on:
+  push:
+    branches:
+      - main
```
Comment (beeme1mr) [inline .github/workflows/pr-checks.yml:6]: Do you think this needs to run when pushing to main?

Comment (skyerus) [reply .github/workflows/pr-checks.yml]: Thanks, removed

**Review by james-milligan — APPROVED** [review]
The detected breaking change isn't a problem - so we should be all good for this to be merged

**Pull request merged**